.travis.yml

```diff
@@ -8,27 +8,45 @@ cache:
 
 php:
     - 5.5
-    - 5.6
+    - 7.0
     - nightly
     - hhvm
 
 matrix:
-    allow_failures:
-        - php: nightly
     fast_finish: true
     include:
         - php: 5.6
-          env: DEPENDENCIES=dev
+          env: SYMFONY_VERSION=2.7.*
+        - php: 5.6
+          env: SYMFONY_VERSION=2.8.*@dev
+        - php: 5.6
+          env: SYMFONY_VERSION="3.0.*@dev"
+        - php: 7.0
+          env: STYLE_CHECK=true
 
-before_script:
-    - composer self-update
-    - if [ "$TRAVIS_PHP_VERSION" = "5.6" ] && [ "$DEPENDENCIES" != "dev" ]; then wget http://get.sensiolabs.org/php-cs-fixer.phar; fi;
-    - if [ "$DEPENDENCIES" = "dev" ]; then perl -pi -e 's/^}$/,"minimum-stability":"dev"}/' composer.json; fi;
+    allow_failures:
+        - php: 7.0
+        - php: nightly
+        - env: SYMFONY_VERSION=2.8.*@dev
+        - env: SYMFONY_VERSION="3.0.*@dev"
+
+before_install:
+  - composer self-update
+  - if [ "$SYMFONY_VERSION" != "" ]; then
```

Comment:
after the `self-update` we could add a `composer config -g github-oauth.github.com $GITHUB_OAUTH_TOKEN` to authenticate against GitHub before update the dependencies. @dunglas this only requires you to add a token with the key `GITHUB_OAUTH_TOKEN`, no permission required (we only need the authentication)